docs/javascript/v2/plugins/noise-suppression.mdx

```diff
@@ -0,0 +1,163 @@
+-   Noise suppression has limited support on Firefox with bluetooth devices.
+
+## Pre-requisites
+
+**Get the 100ms NoiseSuppression Package**
+
+```bash
+npm install --save @100mslive/hms-noise-suppression
+```
+
+## Import plugin
+
+```jsx
+import HMSNoiseSuppressionPlugin from '@100mslive/hms-noise-suppression';
+```
+
+## Instantiate Noise Suppression
+
+This accepts durationInMs as a parameter
+- `durationInMs` - (optional)It maps to the audio samples bufferSize we need to process, by default we are using 80ms equivalent to bufferSize=4096
+
+```jsx
+const noiseSuppressionPlugin = new HMSNoiseSuppressionPlugin(durationInMs);
+```
+
+## Interfaces
+
+### Check if plugin is supported
+
+validateAudioPluginSupport can be used to check if the browser/ input device is supported or not.
+This accepts plugin instance as a parameter
+It will return True in case of plugin is supported and return error message if it is not.
+
+```jsx
+import { hmsActions } from './hms';
+
+const pluginSupport = hmsActions.validateAudioPluginSupport(
+        noiseSuppressionPlugin
+      );
+if (pluginSupport.isSupported) {
+      console.log('Plugin is supported');
+} else{
+      console.log('Plugin is not supported');
+      const err = pluginSupport.errMsg;
+      console.error(err);
+ }
+```
+### Init(Optional)
+
+Init is used to load model of background noise suppression for the first time.
+It takes on an average ..50.. milliseconds.<br/>
+Calling init is handled internally by SDK if not done by user, in this case addPlugin call will
+take around 50 milliseconds for the first time and then less than 1 milliseconds in the subsequent calls.
+Check [this section](#start-and-stop-noise-suppression) for addPlugin API usage
+
+```jsx
+noiseSuppressionPlugin.init();
+```
+
+### Add and Remove Background Noise Suppression
+
+<Tabs id="plugin-code" items={['JavaScript', 'React']} />
+
+<Tab id='plugin-code-0'>
+
+```js
+import { hmsActions, hmsStore } from './hms';
+import { selectIsLocalAudioPluginPresent } from '@100mslive/hms-video-store';
+
+async function toggleNoiseSuppression() {
+  const isNoiseSuppressed = hmsStore.getState(selectIsLocalAudioPluginPresent(noiseSuppressionPlugin.getName()));
+  try {
+    if (!isNoiseSuppressed) {
+      // add background noise suppression
+      await hmsActions.addPluginToAudioTrack(noiseSuppressionPlugin);
+    } else {
+      // remove background noise suppression
+      await hmsActions.removePluginFromAudioTrack(noiseSuppressionPlugin);
+    }
+  } catch (err) {
+    console.log('noise suppression failure - ', isNoiseSuppressed, err);
+  }
+}
+```
+
+</Tab>
+
+<Tab id='plugin-code-1'>
+
+```jsx
+import {
+  useHMSStore,
+  useHMSActions,
+  selectIsLocalAudioPluginPresent
+} from '@100mslive/hms-video-react';
+
+function NoiseSuppression() {
+  const isNoiseSuppressed = useHMSStore(selectIsLocalAudioPluginPresent(noiseSuppressionPlugin.getName());
```

Comment:
add how noiseSuppressionPlugin is created here as well. People will copy the code

Reply:
This is the format we are following for VB also, having a separate section for instantiante plugin and add plugin.

Reply:
We have already added a section for Instantiate plugin above, and that instance is used everywhere